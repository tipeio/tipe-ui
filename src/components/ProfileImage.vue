<template>
  <button data-tipe-ui="TipeProfileImage">
    <TipeImage
      :url="url"
      class="image"
      alt="user profile image"
    />
  </button>
</template>

<script>
import vueTypes from 'vue-types'
import TipeImage from './Image'

export default {
  name: 'TipeProfileImage',
  components: {
    TipeImage
  },
  props: {
    url: vueTypes.string.def('http://placekitten.com/50/50')
  },
  computed: {
    style() {
      return {
        backgroundImage: `url(${this.url})`
      }
    }
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeProfileImage'] {
  border: none;
  padding: 0;
  margin: 0;
  height: 1.6rem;
  width: 1.6rem;
}

.image {
  border-radius: 100%;
  border: none;
  box-shadow: 0px 0px 7px 0px rgba(0, 0, 0, 0.1);
  background: no-repeat center;
  background-size: cover;
}
</style>
