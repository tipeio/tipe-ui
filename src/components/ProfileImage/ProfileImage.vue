<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click')"
  >
    <TipeImage
      :url="url"
      class="image"
      alt="user profile image"
    />
  </button>
</template>

<script>
import vueTypes from 'vue-types'
import TipeImage from '../Image'

export default {
  name: 'TipeProfileImage',
  components: {
    TipeImage
  },
  props: {
    url: vueTypes.string
  },
  computed: {
    style() {
      return {
        backgroundImage: `url(${this.url})`
      }
    }
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeProfileImage'] {
  border: none;
  padding: 0;
  margin: 0;
  height: 3.125rem;
  width: 3.125rem;
  background-color: transparent;
  border-radius: 100%;
  box-shadow: 0 0.125rem 0.875rem 0 rgba(0, 0, 0, 0.3);
  overflow: hidden;
}

.image {
  border: none;
  background: no-repeat center;
  background-size: cover;
}
</style>
