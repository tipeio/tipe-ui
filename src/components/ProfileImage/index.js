export { default } from './ProfileImage.vue'
