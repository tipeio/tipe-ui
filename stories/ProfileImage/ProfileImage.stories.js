import { storiesOf } from '@storybook/vue'
import { withKnobs, text } from '@storybook/addon-knobs/vue'
import { withMarkdownNotes } from '@storybook/addon-notes'

import TipeProfileImage from '../../src/components/ProfileImage'

const notes = `
#### :props
:url | string |
`

storiesOf('Profile Image', module)
  .addDecorator(withKnobs)
  .add(
    'has profile image',
    withMarkdownNotes(notes)(() => ({
      components: { TipeProfileImage },
      data() {
        return {
          url: text('url', 'http://placekitten.com/200/300')
        }
      },
      template: `
      <tipe-profile-image
        :url="url"
      />`
    }))
  )
