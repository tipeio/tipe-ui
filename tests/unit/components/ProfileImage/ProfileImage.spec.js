import { shallowMount } from '@vue/test-utils'
import ProfileImage from '@/components/ProfileImage'
import TypeImage from '@/components/Image'
import faker from 'faker'

describe('ProfileImage', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(ProfileImage)

      expect(wrapper).toMatchSnapshot()
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(ProfileImage)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeProfileImage')
    })
  })

  describe(':props', () => {
    it('url should default to http://placekitten.com/200/300', () => {
      const wrapper = shallowMount(ProfileImage)

      expect(wrapper.props().url).toEqual('http://placekitten.com/200/300')
    })

    it('should pass url prop to Image', () => {
      const propsData = { url: faker.internet.url() }
      const wrapper = shallowMount(ProfileImage, { propsData })
      const imageWrapper = wrapper.find(TypeImage)

      expect(imageWrapper.props().url).toEqual(propsData.url)
    })
  })

  describe('@events', () => {
    it('click', () => {
      const wrapper = shallowMount(ProfileImage)

      wrapper.trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
