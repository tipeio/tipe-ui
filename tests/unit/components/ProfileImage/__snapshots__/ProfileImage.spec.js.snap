// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage <template> matches previous snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
  "version": 2.5,
}
`;
